visualization/app/codeCharta/util/gameObjectsParser/gameObjectsImporter.ts

```diff
@@ -0,0 +1,138 @@
+import md5 from "md5"
+
+function isFile(names) {
+	return names.length === 0
+}
+
+function nodeAlreadyExists(nodes, name) {
+	return nodes.some(node => node.name === name)
+}
+
+function addNodeRecursively(names, nodes, gameObjectPosition, parentNodeName, gameObjectPositions) {
+	if (names.length === 0) return
+	// get current path name
+	const name = names.shift()
```

Comment:
```suggestion
	const [name] = names
```
You should check out array destructuring! This retrieves the first element WITHOUT mutating the original array. `names` is passed down again later one and it's not that obvious, that you removed the first element here. In general it's not a good idea modify function parameters. This can result in really nasty bugs.
Instead create new arrays and pass them down. I'd do this call at the end then:

```
	addNodeRecursively(names.slice(1), node.children, gameObjectPosition, newParentName, gameObjectPositions)
```

Reply:
That wouldn't have occurred to me in a million years. Thank you for your input!
Will fix it in the next commit

Reply:
Hey, a short follow-up question. The rest of the method relies on the array becoming shorter for other checks. So after taking the first element it shouldn't be in the array anymore. Should we do `const [name] = names; names = names.slice(1)` But isn't that the same as `const name = names.shift()`? Or should I copy it and work with a copy?